front/src/app/shared/task-list-manager/task-list-manager.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { TaskGroupsList } from '../models/shared.models';
+import { ApiService } from '../api-service/api.service';
+import { BehaviorSubject, Observable, Subject, map, takeUntil } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskListManagerService {
+
+  listId$: Observable<string> = this.route.queryParams.pipe(
+    map((params): string => params['list']),
+  );
+
+  private allTaskListsSubject$ = new BehaviorSubject<TaskGroupsList | undefined>(undefined)
+
+  private selectedTaskListIndexSubject$ = new BehaviorSubject<number | undefined>(undefined)
+
+  private destroy$ = new Subject<void>();
+
+  get allTaskLists$(): Observable<TaskGroupsList | undefined> {
+    return this.allTaskListsSubject$.asObservable();
+  }
+
+  get selectedTaskListIndex$(): Observable<number | undefined> {
+    return this.selectedTaskListIndexSubject$.asObservable();
+  }
+
+  constructor(private apiService: ApiService, private router: Router, private route: ActivatedRoute) { }
+
+  loadLists(): void {
+    this.apiService.getLists()
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(data => {
+        this.allTaskListsSubject$.next(data)
+      });
+  }
+
+  deleteTaskListById(id: string): void { 
+    this.apiService.deleteTaskList(id)
+      .subscribe(() => {
+        this.router.navigate([]);

```

Comment:
Is there a need to navigate to another page if some list was deleted? (what if deleted task is not currently opened one?)